src/tokens.js

```diff
@@ -112,45 +112,72 @@ const tokens = {
    * @param {string} config Token configuration string
    * @param {string} uid Uid to check if matches with uid from config (optional)
    *
-   * @return {Object} {success: boolean, message: in case of failure, tokenData: object with token data in case of success}
+   * @return {Promise} Promise that resolves when validation finishes. Resolves with tokenData {uid, name, symbol} and reject with error message
    *
    * @memberof Tokens
    * @inner
    */
   validateTokenToAddByConfigurationString(config, uid) {
-    const tokenData = this.getTokenFromConfigurationString(config);
-    if (tokenData === null) {
-      return {success: false, message: 'Invalid configuration string'};
-    }
-    if (uid && uid !== tokenData.uid) {
-      return {success: false, message: `Configuration string uid does not match: ${uid} != ${tokenData.uid}`};
-    }
+    const promise = new Promise((resolve, reject) => {
+      const tokenData = this.getTokenFromConfigurationString(config);
+      if (tokenData === null) {
+        reject('Invalid configuration string');
+      }
+      if (uid && uid !== tokenData.uid) {
+        reject(`Configuration string uid does not match: ${uid} != ${tokenData.uid}`);
+      }
 
-    const validation = this.validateTokenToAddByUid(tokenData.uid);
-    if (validation.success) {
-      return {success: true, tokenData: tokenData};
-    } else {
-      return validation;
-    }
+      const promiseValidation = this.validateTokenToAddByUid(tokenData.uid, tokenData.name, tokenData.symbol);
+      promiseValidation.then(() => {
+        resolve(tokenData);
+      }, (message) => {
+        reject(message);
+      });
+    });
+    return promise;
   },
 
   /**
-   * Validation token by uid. Check if already exist
+   * Validation token by uid.
+   * Check if this uid was already added, if name and symbol match with the information in the DAG,
+   * and if already have another token with this name or symbol already added
    *
    * @param {string} uid Uid to check for existence
+   * @param {string} name Token name to execute validation
+   * @param {string} symbol Token symbol to execute validation
    *
-   * @return {Object} {success: boolean, message: in case of failure}
+   * @return {Promise} Promise that will be resolved when validation finishes. Resolve with no data and reject with error message
    *
    * @memberof Tokens
    * @inner
    */
-  validateTokenToAddByUid(uid) {
-    const existedToken = this.tokenExists(uid);
-    if (existedToken) {
-      return {success: false, message: `You already have this token: ${uid} (${existedToken.name})`};
-    }
+  validateTokenToAddByUid(uid, name, symbol) {
+    const promise = new Promise((resolve, reject) => {
+      // Validate if token uid was already added
+      const existedToken = this.tokenExists(uid);
```

Comment:
existedToken?

Reply:
Reading again it seems a bad var name, changed to `token`. Thanks